### ISSUE TITLE
Issue with URL Rewriting ColdFusion 11/Apache2.4/Ubuntu

### ISSUE THREAD

**Issue by jkeczan** [cfwheels/cfwheels#367]
I have URL Rewriting turned on and it is working all for except one case. When I access my base url:

http://sub.domain.com

All of my links are built like the following:

http://sub.domain.com/index.cfm/controller

Since the link is built with index.cfm and URL Rewriting is turned on, CFWheels thinks the controller is indexcfm. This causes my security script that verifies user access by controller to fail. Are there any thoughts on this one?

My .htaccess files
```
Options +FollowSymLinks
RewriteEngine On
RewriteCond %{REQUEST_URI} !^.*/(503|flex2gateway|jrunscripts|cfide|cfformgateway|cffileservlet|railo-context|files|images|javascripts|miscellaneous|stylesheets|robots.txt|sitemap.xml|rewrite.cfm|remote)($|/.*$) [NC]$
RewriteRule ^(.*)$ ./rewrite.cfm/$1 [NS,L]
```

**Comment by jkeczan** [comment]
Just wanted to update you guys. The fix for this ended up being in the .htaccess file. we added an additional rule for the base case.
```
RewriteCond %{REQUEST_URI} ^.*/index.cfm/(.*)$ [NC]
RewriteRule ^.*/index.cfm/(.*)$ ./rewrite.cfm/$1 [NS,L]

RewriteCond %{REQUEST_URI} !^.*/(flex2gateway|jrunscripts|fonts|cfide|cfformgateway|cffileservlet|railo-context|files|images|javascripts|miscellaneous|stylesheets|robots.txt|favicon.ico|sitemap.xml|rewrite.cfm|remote)($|/.*$) [NC]
RewriteRule ^(.*)$ ./rewrite.cfm/$1 [NS,L]
```


This works for Ubuntu 14.04, Apache 2.4, CF 11

**Comment by perdjurner** [comment]
Thanks for the update.

Do you know why it does not get picked up by the Wheels RewriteCond / RewriteRule?
It seems like it should have...

Maybe I am missing something obviously wrong with the Wheels rewrite rules?

**Comment by dbelanger** [comment]
Per, I just upgraded from 1.1.8 to 1.3.3 and I can confirm that what is happening to jkeczan is also happening to me.  If I comment out the .htaccess rules, I can navigate the site without issue ("index.cfm/controller/action/key" is shown) but the minute I un-comment the rules, I get the same error "Could not find the view page for the index action in the Rewritecfm controller".  The only thing I changed from 1.1.8 to 1.3.3 was exactly what was specified in the upgrade document - the Wheels directory and the root.cfm file.

I also ran the tests you specified above (by placing the <cfabort> in the rewrite.cfm file and it's gets aborted every time so we know that the .htaccess rules are working.  I'm running Ubuntu 13.04, CF10SP13, Apache 2.2.22.

**Comment by dbelanger** [comment]
I found it Per!

In your wheels/dispatch/request.cfm file ($request function), the following:

	<cfargument name="pathInfo" type="string" required="false" default="#replacenocase(request.cgi.path_info,'rewrite.cfm/','')#">

was changed to

<cfargument name="pathInfo" type="string" required="false" default="#request.cgi.path_info#">

I changed it back and "voila", right back to normal!

**Comment by perdjurner** [comment]
Good find but I downloaded 1.1.8 and it does not have the replacenocase code in it actually. So, you must have added it yourself at some point? Do you remember why, where you got it from etc?

**Comment by dbelanger** [comment]
Sorry for the delay Per.  

That particular piece of code that I gave you was inserted into the project before I arrived and I only saw it when I updated the Wheels core to 1.3.3.  I just assumed it was part of the original wheels code.

From what I can see, the previous developer was having the same issues I was where the rewrite.cfm string wasn't being removed in the redirect and added this code as a backup.  This looks to be more because of Ubuntu/Apache than Wheels.

We've just upgraded our local dev environment to Ubuntu 14.04.2 and Apache 2.4.7 (CF10SP15).  The additional string mentioned by jkeczan in the .htaccess file works great in that environment.

While leaving in the code I mentioned doesn't hurt anything, it doesn't look like it's required anymore.  I would recommend closing the ticket again.

**Comment by perdjurner** [comment]
Thanks for the update.

I will leave this open since the .htaccess file is also part of Wheels and it might need to be corrected.

**Comment by perdjurner** [comment]
The following is a possible fix:
https://github.com/cfwheels/cfwheels/commit/6b8ddb0bf271c1e3b239e8aad3f1a13519409ece

Please reopen if not (you can either try it now or wait until the 1.4 release).

**Comment by dbelanger** [comment]
I just installed and am testing 1.4.0 and I'm still getting the same issue mentioned in this issue (URL Rewriting on CF11).  Wheels is still looking for the action in "rewritecfm" controller (notice the period was removed but not "rewrite" or "cfm".  As I mentioned before, changing the default of the pathInfo argument on line 175 of the dispatch/request.cfm to the code below fixes the issue.

`default="#replacenocase(request.cgi.path_info,'rewrite.cfm/','')#"`

I'm on Ubuntu 14.04.2, Apache 2.4.7, CF11 Update 5
